src/coreclr/ilasm/asmparse.y

```diff
@@ -263,11 +263,11 @@ int32                   : INT32_V                             { $$ = $1; }
                         ;
 
 int64                   : INT64_V                             { $$ = $1; }
-                        | INT32_V                             { $$ = neg ? new __int64($1) : new __int64((unsigned)$1); }
```

Comment:
Are there any plans to also fix up casts like `unsigned` to be `uint32_t` (in a future PR)?

Reply:
The classic C types like `unsigned` are used 1000+x more frequently than the MS specific types that I am cleaning up here. I do not plan to replace them. I am not sure whether bulk find&replace for those would be a net improvement, it would be much more disruptive, and potentially breaking due to subtle C/C+ overload resolution rules.

My motivation for this PR: I got annoyed by the types with double underscores around code changed in #102225. I have noticed that there are not that many places where we use them, so I have done this find&replace (+revert where it caused build breaks)